WordPress/gutenberg
# List View: Add multi-select support for shift + Home and End keys

## Description


Following on from https://github.com/WordPress/gutenberg/pull/38314, this PR adds in support for shift-selecting multiple blocks within the list view via the Home and End keys. The Tree Grid component already supports navigating to the beginning and end of the block list via these keys — this PR is a fairly simple update that rolls out adding in the `onFocusRow` callback for those key press events, too, and accepting those events as valid key presses for multiple block selection in `useBlockSelection`.

The end result is that it should now be a little easier to select from a particular block within the list view, to the end or beginning of a post.

## Testing Instructions





1. Add a range of blocks to a post. Open up the list view, and tab to focus on a block within the list.
2. Use Shift + Home or Shift + End keys to select a range of blocks.
3. Also use Shift + Up or Shift + Down keys to adjust the selection.
4. Check that this doesn't introduce any regressions.

Check that the added tests pass in CI (or you can use the following to run locally):

```
npm run test-unit packages/components/src/tree-grid/test/index.js
```

## Screenshots 

In the following screengrab, the Home and End keys allow selection to the beginning or end of the list while the shift key is selected. The up and down arrows then still enable altering that selection.

![list-view-home-and-end-keys](https://user-images.githubusercontent.com/14988353/157146280-b8c9e138-8d8b-47a3-bb13-ace8eaf4362e.gif)

## Types of changes




Enhancement.

## Checklist:
